src/devices/Common/Iot/Device/Common/WeatherHelper.cs

```diff
@@ -0,0 +1,210 @@
+
+            if (steadman + tf < 160) // if the average is lower than 80F, use Steadman, otherwise use Rothfusz.
+                return Temperature.FromFahrenheit(steadman);
+
+            double rothfuszRegression = (-42.379)
+                + (2.04901523 * tf)
+                + (10.14333127 * rh)
+                - (0.22475541 * tf * rh)
+                - (6.83783 * 0.001 * tf2)
+                - (5.481717 * 0.01 * rh2)
+                + (1.22874 * 0.001 * tf2 * rh)
+                + (8.5282 * 0.0001 * tf * rh2)
+                - (1.99 * 0.000001 * tf2 * rh2);
+
+            if (rh < 13 && tf >= 80 && tf <= 112)
+            {
+                rothfuszRegression += ((13 - rh) / 4) * Math.Sqrt((17 - Math.Abs(tf - 95)) / 17);
+            }
+            else if (rh > 85 && tf >= 80 && tf <= 87)
+            {
+                rothfuszRegression += ((rh - 85) / 10) * ((87 - tf) / 5);
+            }
+
+            return Temperature.FromFahrenheit(rothfuszRegression);
+        }
+
+        /// <summary>
+        /// The summer simmer index is used to measure the amount of discomfort during the summer months
+        /// when heat and humidity often combine to make it feel hotter than it actually is. The summer
+        /// simmer index is usually used for overnight low temperatures. Use the heat index instead for
+        /// afternoon high temperatures.
+        /// </summary>
+        /// <param name="airTemperature">The dry air temperature</param>
+        /// <param name="relativeHumidity">The relative humidity (RH) expressed as a percentage</param>
+        /// <returns>The summer simmer index</returns>
+        public static Temperature CalculateSummerSimmerIndex(Temperature airTemperature, double relativeHumidity)
+        {
+            double tf = airTemperature.Fahrenheit;
+            double rh = relativeHumidity;
+            return Temperature.FromFahrenheit(1.98 * (tf - ((0.55 - (0.0055 * rh)) * (tf - 58.0))) - 56.83);
+        }
+
+        /// <summary>
+        /// Calculates the saturated vapor pressure.
+        /// </summary>
+        /// <param name="airTemperature">The dry air temperature</param>
+        /// <returns>The saturated vapor pressure</returns>
+        public static Pressure CalculateSaturatedVaporPressure(Temperature airTemperature)
+        {
+            double tc = airTemperature.Celsius;
+            return Pressure.FromHectopascal(6.11 * Math.Pow(10, ((7.5 * tc) / (237.7 + tc))));
+        }
+
+        /// <summary>
+        /// Calculates the actual vapor pressure.
+        /// </summary>
+        /// <param name="airTemperature">The dry air temperature</param>
+        /// <param name="relativeHumidity">The relative humidity (RH) expressed as a percentage</param>
+        /// <returns>The actual vapor pressure</returns>
+        public static Pressure CalculateActualVaporPressure(Temperature airTemperature, double relativeHumidity)
```

Comment:
What does actual mean here? Does it mean current? or Relative? Not sure I understand the difference between this API and the previous one.

Reply:
"Actual vapor pressure" is a term. http://ww2010.atmos.uiuc.edu/(Gh)/guides/mtr/cld/dvlp/rh.rxml#targetText=Actual%20vapor%20pressure%20is%20a,amount%20of%20water%20vapor%20increases.&targetText=That%20means%2C%20an%20equal%20number,air%20back%20into%20the%20water.

Reply:
I see, I didn't know that 😄

NIT: Any chance we can add that description in the summary so that it is more clear for people? so from that link you sent, it would be something like:
```c#
/// Calculates the actual vapor pressure, which is a measurement of the amount of water vapor in a volume of air and increases as the amount of water vapor increases.
```

If you think that most people using this method will know what actual vapor pressure it means then I'm totally fine keeping this as is, but if not I guess we could use the extra description here (I'm sorry for being picky 😞)